core/domain/user_services.py

```diff
@@ -111,14 +111,30 @@ def get_multi_user_ids_from_usernames(usernames):
         usernames: list(str). Identifiable usernames to display in the UI.
 
     Returns:
-        list(str). Return the list of user ids corresponding to given
+        list(str|None). Return the list of user ids corresponding to given
         usernames.
         """
     user_ids: List[str] = []
-    for username in usernames:
-        user_id = get_user_id_from_username(username)
-        if user_id is not None:
-            user_ids.append(user_id)
+    normalized_usernames = [
+        user_domain.UserSettings.normalize_username(username)
+        for username in usernames
+    ]
+    found_models = user_models.UserSettingsModel.query(
```

Comment:
With get_multi(), you have guarantee of order, but I don't think you have that for an IN query, so you probably will need to make a dict of usernames to user ID and use that to construct the return list.

Reply:
Hmm, have updated this and added a dict before creating the final list.

Reply:
Looks good. Thanks!